### ISSUE TITLE
Portable Version?

### ISSUE THREAD

**Issue by war59312** [OpenLiveWriter/OpenLiveWriter#113]
Hey,

Should would be nice to have a portable version.

I would love to see a PortableApps.com build.

Thanks,

Will

**Comment by amrta** [comment]
I just want to say this.

**Comment by paulcbetts** [comment]
While it's not Official, taking `C:\Users\USER\AppData\Local\OpenLiveWriter` and copying that whole folder to a USB key will Just Work, including app updates

**Comment by ScottIsAFool** [comment]
@paulcbetts For the most part, yes, but there are things that wouldn't get copied across, like blog templates, glossary links etc, which all reside in %appdata%\OpenLiveWriter\

**Comment by martinwoodward** [comment]
What's the preferred approach to configure a portable app's settings?  We could pretty easily have the settings be configured by a .exe.config if that was good enough. Would we need to persist the settings that are currently stored in the registry to a configurable file as well?

**Comment by ScottIsAFool** [comment]
@martinwoodward correct, registry settings would have to be exported too.

**Comment by paulcbetts** [comment]
@martinwoodward The way that Atom does it (designed by yours truly), is the following:

1. Look in `%AppData%\OpenLiveWriter` for stuff. Is it there? Use it!
1. Are we outside of the default installation directory? If so, the App Data folder is `Path.Combine(DirectoryOfMyExeFile, "..", "AppData")` or something to that effect

**Comment by martinwoodward** [comment]
Thanks @paulcbetts

**Comment by gep13** [comment]
@martinwoodward if you end up going down the route of a portable version of the application, then we can look to extending the chocolatey package to include:

* OpenLiveWriter
* OpenLiveWriter.install
* OpenLiveWriter.portable

to allow people to have a choice about what they want to install.

**Comment by Gordon-Beeming** [comment]
Is anyone working on switching the settings to be File system based? Can I work on this? 

It would be nice to make this so that we could potentially have this easily use other sources for settings. I have some code currently that I could adapt to OLW that stores settings in JSON. 

We could make the settings by default use the folder locations suggested above and then have a setting some time in the future allowing people to choose a path for the settings and have only the main setting of where is that stored in the paths mentioned above and potentially have the settings live in a OneDrive folder or Dropbox to easily sync between machines.

Would make use of factories so we could easily add support for writing settings directly to some online storage (as a possible place for people to make an add on).

Just some thoughts.

**Comment by ScottIsAFool** [comment]
Go for it :)

**Comment by Gordon-Beeming** [comment]
awesome =).

**Comment by Gordon-Beeming** [comment]
It looks like OLW already has code for a portable version.

https://github.com/OpenLiveWriter/OpenLiveWriter/blob/master/src/managed/OpenLiveWriter.CoreServices/ApplicationEnvironment.cs#L78

Can anybody confirm how 'stable' or usable this is =)

**Comment by martinwoodward** [comment]
Wow, look at that. Welcome to the world of Live Writer Gordon! Looks like a partly implemented feature that just needs testing and finished off. We'd need to generate a (signed) portable build exe as part of the build process but looking promising that it won't be too much work. Should a portable app update itself using squirrel or is that not good form?

**Comment by gep13** [comment]
We could at a openlivewriter.portable package to chocolatey, and allow updates through that. Thoughts?

Sent from my Windows Phone
________________________________

**Comment by paulcbetts** [comment]
@gep13 Doesn't sound very portable!

**Comment by johnstrasser** [comment]
Hi Folks,

Fwiw, all I'd ask for is a zip file that let's me copy everything to a directory of my choice, and stores all the settings in that directory.  Kinda like eclipse (eclipse . org) and todolist (abstractspoon . com).

And the install can be just as easy, no need for anything else.  Once upon a time this was doable. Is that what you're all talking about?

Thanks

JS

**Comment by vatterspun** [comment]
Just a quick addition here: portability means a lot of things to a lot of people but just to be clear: most users are looking for the program to save any and all settings to the local folder.

**Who will benefit from this?**

* Cloud capability - An individual who has OpenLiveWriter stored on their cloud drive (Dropbox, Google Drive, etc.) and wants to have the same experience/settings from one machine to another
* USB drive - Someone who uses a public terminal at a coffee shop, computer lab, etc.
* Ease of backup / encryption (e.g. Just quickly a folder rather than trying to track down relevant files in in the My Documents folder; storing the program inside a TrueCrypt / VeraCrypt volume)
* Mac/Linux users (via WINE/Crossover Office) - lets users just use the Windows version rather than re-entering their config across systems

**Other notes**

As this program is fairly old and has a solid following, some users will invariably prefer the old behavior as it works better with their existing backup/restore system. As a result, I recommend making this optional.

[Note: this was pulled from a related issue post in the interests of keeping everything in one place.  Not trying to spam the site with this issue..]

**Comment by kiquenet** [comment]
Portable version would be fantastic and marvellous.

Now, in a company  I'm NOT local administrator in my laptop. I cannot install anything. 

Only network admin can install programs in my laptop.

I  use Notepad++ portable.

I need Open Live Writer portable.

**Comment by lon3wolf** [comment]
One can make a portable app, by handling all the settings - (registry, FS calls etc.) and redirecting them to virtualized store
Something like [this](http://portableapps.com/apps/development/portableapps.com_launcher) can be used to make it portable (no coding required).

Other option is Virtualization

**Comment by kiquenet** [comment]
Not useful. I cannot install PortableApps for create portable version neither install HyperV, VMWare, VirtualBox...

**Comment by vatterspun** [comment]
> Something like this [PortableApps.com] can be used to make it portable (no coding required).

That's true and there are actually a variety of wrapper programs that will do what PA will do.  The catch is that PA won't actually add the program to their site database because it's .NET based.  So you can build one but you get to also maintain it.  Updates in the future might break the PA wrapper whereas programs that are natively portable don't have this issue.

**Comment by r-a-y** [comment]
> 
> 
> It looks like OLW already has code for a portable version.
> 
> https://github.com/OpenLiveWriter/OpenLiveWriter/blob/master/src/managed/OpenLiveWriter.CoreServices/ApplicationEnvironment.cs#L78
> 
> Can anybody confirm how 'stable' or usable this is =)

I just tested this. It's quite portable.

To set up portable mode:

1.  Download `OpenLiveWriterSetup.exe` from the [website](https://openlivewriter.com/) and open it in 7-zip, then extract the `.nupkg` file. (Alternatively, you can also get the latest .nupkg development version [here](https://ci.appveyor.com/api/buildjobs/8vpaqgg6nfihmlye/artifacts/Releases%2FOpenLiveWriter-0.6.3-full.nupkg).)
2. Next, open the `.nupkg` file with 7-Zip and extract the contents of `lib\net451` to somewhere like `c:\myapps\openlivewriter`. 
3. In `c:\myapps\openlivewriter`, create a new folder called `UserData`.
4. Open `OpenLiveWriter.exe` and that should be it!

**Comment by rkingett** [comment]
It works! I still think it would be worth it to make an official PortableApps version though, but how well does their system work with github? Would a developer have to manually update the portableapps version every time? I'm eagerly awaiting on the new accessible version for screen readers, personally, but the directions above work great

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
